extensions/mssql/src/hdfs/webhdfs.ts

```diff
@@ -356,7 +356,21 @@ export class WebHDFS {
 				callback(error, undefined);
 			} else if (response.body.hasOwnProperty('FileStatuses')
 				&& response.body.FileStatuses.hasOwnProperty('FileStatus')) {
-				files = response.body.FileStatuses.FileStatus;
+				files = (<any[]>response.body.FileStatuses.FileStatus).map(fs => {
+					return new FileStatus(
```

Comment:
Is there any property we might be missing here? I am wondering if we risk losing some things here that we'd have had in the past. Otherwise, this is a good change.

Reply:
So the return value from the call had these turned into IHdfsFileStatus (the interface I removed from fileSources). This interface only had two properties, type and pathSuffix. So yes - while the actual objects might have other properties on them the typing didn't say that and so I don't see why anyone would be trying to use other properties. So really this is adding a whole bunch of other properties that we didn't have before.